### ISSUE TITLE
Cloudflare configuration page Blank

### ISSUE THREAD

**Issue by erik921** [cloudflare/Cloudflare-Magento#56]
Hello,

I added the cloudflare magento 2 plugin to my magento store.
It is running magento 2.1.6 and the cloudflare plugin is also running the latest version (just updated it using composer)

When I visit the configuration page the page is blank. Looking at the console in chrome I get this error:

compiled.min.js:13 Uncaught ReferenceError: ε is not defined
    at compiled.min.js:13
    at Object.<anonymous> (compiled.min.js:13)
    at __webpack_require__ (compiled.min.js:1)
    at compiled.min.js:18
    at Object.<anonymous> (compiled.min.js:18)
    at __webpack_require__ (compiled.min.js:1)
    at Object.<anonymous> (compiled.min.js:13)
    at __webpack_require__ (compiled.min.js:1)
    at Object.<anonymous> (compiled.min.js:18)
    at __webpack_require__ (compiled.min.js:1)

Thanks in Advance!
Erik

**Comment by jwineman** [comment]
Hi @erik921,

If you run
```
bin\magento cache:clean
bin\magento cache:flush
```
Does the problem persist?  

When you say "the configuration page" you mean the Cloudflare plugin index page correct? Can you send a screenshot of the page?

Thanks,
John

**Comment by erik921** [comment]
Hello John,

Correct.
I also tried running like the full stack so:

bin/magento setup:static-content:deploy
bin/magento setup:di:compiler
bin/magento indexer:reindex
bin/magento cache:clean
bin/magento cache:flush

Correct I mean the Cloudflare plugin index page. Sorry for being so confusing.
Here is the screenshot you asked for: http://prntscr.com/f0pd8a

Yours sincerely,
Erik

**Comment by jwineman** [comment]
What other extensions do you have installed on Magento?

**Comment by erik921** [comment]
Currently using:
-Venusbrand
-Apptrain Image optimser
-Myparcel
-Bss one step checkout
-Bss configurable grid view
-Magemonkey
-Channable
-Searchsuite

**Comment by jwineman** [comment]
My assumption is that one of the other plugins is bundling their Javascript with webpack and the loaders are being overwritten, I'm not sure what the fix is though.

**Comment by epicglows** [comment]
Same problem here on M2.1.6:

Uncaught ReferenceError: ε is not defined
    at compiled.min.js:13
    at Object.<anonymous> (compiled.min.js:13)
    at __webpack_require__ (compiled.min.js:1)
    at compiled.min.js:18
    at Object.<anonymous> (compiled.min.js:18)
    at __webpack_require__ (compiled.min.js:1)
    at Object.<anonymous> (compiled.min.js:13)
    at __webpack_require__ (compiled.min.js:1)
    at Object.<anonymous> (compiled.min.js:18)
    at __webpack_require__ (compiled.min.js:1)

Were you able to get it fixed @erik921 ?
Any fix @jwineman ?
Thanks!

**Comment by erik921** [comment]
hey @epicglows 
No the issue is still there for me even with the new update for both the plugin and magento.
I'm going to try to solve it but currently have some things on my to do list that I need to do first.

**Comment by lingwooc** [comment]
Same here on M2.1.8.

@jwineman It is related to Advanced > Developer > Minify JavaScript Files. Turn that off and the page loads OK. Stack exchange suggests the following fix (I've not tried it myself) https://magento.stackexchange.com/questions/162624/exclude-custommodule-js-from-minification-of-js-in-magento2

Assuming the module doesn't add some similarly afflicted js to normal pages we should be able to just turn it off, setup, then turn it back on until the fix comes in.

**Comment by nikonratm** [comment]
Is anyone trying to work on this? Having the same issue

**Comment by lingwooc** [comment]
Did you try my work around?

**Comment by nikonratm** [comment]
@lingwooc Yes your workaround was super helpful and got things working, thanks! 

But obviously re-compiling a live site (twice) every time you need to access CF admin isn't a sustainable solution, I was hoping a more permanent fix is in the works.

**Comment by lingwooc** [comment]
@nikonratm Fair enough. Can't argue with your assessment either. Just wanted to make sure this was everyone's issue not just mine.

**Comment by garrettgalow** [comment]
We have reproduced the issue and are looking into this. Tracked internally as PI-1234

**Comment by gordianberger** [comment]
The file seems to be minified with webpack. Since magento tries to minify it again some bugs will occur. The fix is actually simple: rename the file to combined.min.js and magento won't minify it again. We do this with our own webpack modules as well.

**Comment by garrettgalow** [comment]
If magento requires files to be .min.js when minified won't that choke if the user doesn't turn on minification - aka it would just look for combined.js?

**Comment by bruno-serfe** [comment]
Hey Guys, this is still an issue using latest versions of Magento 2.2.x, will come out a fix or a new version of the extension?

Thanks!

**Comment by garrettgalow** [comment]
There was a delay in getting the package updated on Magento's site and in composer. We should have that resolved soon. Apologies for the trouble

**Comment by bruno-serfe** [comment]
Hi @garrettgalow thanks for the quick response, it seems there's a know bug https://github.com/cloudflare/Cloudflare-Magento/pull/64 and it's hard to resolve, I've tried with this https://magento.stackexchange.com/questions/162624/exclude-custommodule-js-from-minification-of-js-in-magento2 but is not working...
is there a way to pass the "config.js" json options in another way? or to provide the compiled.js code uncompiled? in order to create a new file and provide that "config.js" options file using the magento 2 requirejs-config.js file.

Thanks,

**Comment by garrettgalow** [comment]
we fixed it with #72 so you can see the change there on how we patched it.

**Comment by bruno-serfe** [comment]
@garretgalow I've managed to solve the issue, I've created a module to override CloudFlare_plugin and:

1) I had to modify the file you mention in #72
2) After that since I've not excluded other files like config.js and en.js I had to modify compiled.js file with the following:
   a) Modify the lines where "./config.js" and "/en.js" were called.
   b) And replacing them with "config.min.js" and "en.min.js" respectively.

That works for me using Magento v2.2.5 I hope this will be useful for anyone else.

Btw, thanks for your help!
